### DISCUSSION TITLE
Support Bft unit of measurement for wind speed sensors

### DISCUSSION THREAD

**Discussion by fwestenberg** [home-assistant/architecture#1020] [Entity Models] [answered]
In the current situation, wind speed sensors can be measured in 'ft/s', 'km/h', 'kn', 'm/s', 'mph'. The idea is to add the Beaufort (Bft) scale to wind_speed. This requires a new of conversion, as it is not so straight forward. Also, Bft is not a unit of measurement for speed sensors, it applies only to wind_speed.

m/s to Bft: `round(((ms / 0.836) ** 2) ** (1 / 3))`

Bft to m/s: `0.836 * beaufort ** (3 / 2)`

**Comment by frenck** [comment]
As with any proposal, we need to know which integrations could use this.

Currently, we have an enumeration for the unit of speed, which is used for all wind speeds across the board. As you states: Btf is not an unit of measurement for general speed, but only for measuring the unit of Wind speed.

I guess that means we are introducing a new enumeration and migrating all existing wind sensors onto it?

../Frenck

**Reply by emontnemery** [reply]
I think this would make sense to add support for the Beaufort scale allow UI-configuration of wind speed sensor and weather entities to display wind speed in Beaufort, we don't need existing integrations to produce sensors which report wind speed in Beaufort.

Or to put it in another way: Any weather integration would benefit from this change because users could choose to display wind speed in Beaufort.

**Reply by emontnemery** [reply, marked as answer]
We have these speed related device classes now:
```
SensorDeviceClass.PRECIPITATION_INTENSITY: set(UnitOfVolumetricFlux),
SensorDeviceClass.SPEED: set(UnitOfSpeed).union(set(UnitOfVolumetricFlux)),
SensorDeviceClass.WIND_SPEED: set(UnitOfSpeed),
```

All units in `UnitOfSpeed` are relevant for wind speed sensors, and the catch all `speed` class is the union of all units: there's no need to add another enum now, Beaufort can be added to `UnitOfSpeed`

**Reply by frenck** [reply]
Let's do it 👍

**Comment by fwestenberg** [comment]
This seems the most reasonable solution. 

The integration I am thinking of now, is the Stookwijzer. The website indicated the windspeed in Bft. But also for weather providers, Bft is a common unit of measurement.

**Reply by frenck** [reply]
> The integration I am thinking of now, is the Stookwijzer.

Right, but we don't want to make these changes for a single integration, so this needs research on which other integrations support Bft.

**Comment by gjohansson-ST** [comment]
Implemented in https://github.com/home-assistant/core/pull/105795
